core/domain/exp_domain.py

```diff
@@ -3060,90 +3169,1468 @@ def _convert_states_v51_dict_to_v52_dict(
+        is_tags_nested_inside_tabs_or_collapsible: bool = False
+    ) -> str:
+        """Handles all the invalid RTE tags, performs the following:
+            - `oppia-noninteractive-image`
+                - If `alt-with-value` attribute not in the image tag,
+                introduces the attribute and assign empty value
+                - If `filepath-with-value` attribute not in image tag,
+                removes the tag
+                - If `filepath-with-value` attribute empty then removes
+                the tag
+                - If `caption-with-value` attribute not in the image tag,
+                introduces the attribute and assign empty value
+            - `oppia-noninteractive-skillreview`
+                - If `text-with-value` attribute is not present or empty or
+                None, removes the tag
+                - If `skill_id-with-value` attribute is not present or empty or
+                None, removes the tag
+            - `oppia-noninteractive-math`
+                - If `math_content-with-value` attribute not in math tag,
+                removes the tag
+                - If `raw_latex` is not present or empty or None, removes
+                the tag
+            - `oppia-noninteractive-video`
+                - If `start-with-value` or `end-with-value` is not present,
+                introduce them to the tag and assign 0 to them
+                - If `autoplay-with-value` is not present or is not boolean,
+                introduce it to the tag and assign `false` to them
+                - If `video_id-with-value` is not present or empty, removes
+                the tag
+                - If `start-with-value` > `end-with-value`, set both to '0'
+            - `oppia-noninteractive-link`
+                - If `text-with-value` or `url-with-value` is not present,
+                or is empty simply removes the tag
+            - `oppia-noninteractive-tabs` and `oppia-noninteractive-collapsible`
+                - If these tags are nested inside tabs and collapsible tag, we
+                will simply remove the tag
+
+        Args:
+            html: str. The RTE tags.
+            is_tags_nested_inside_tabs_or_collapsible: bool. If the tag is
+                present inside the tabs or collapsible tag.
+
+        Returns:
+            str. Returns the updated html value.
+        """
+        empty_values = ['&quot;&quot;', '', '\'\'', '\"\"']
+        soup = bs4.BeautifulSoup(html, 'html.parser')
+
+        for tag in soup.find_all('oppia-noninteractive-image'):
+            if not tag.has_attr('alt-with-value'):
+                tag['alt-with-value'] = '&quot;&quot;'
+            if not tag.has_attr('filepath-with-value'):
+                tag.decompose()
+            else:
+                if tag['filepath-with-value'] in empty_values:
+                    tag.decompose()
+            if not tag.has_attr('caption-with-value'):
+                tag['caption-with-value'] = '&quot;&quot;'
```

Comment:
```suggestion
            if not tag.has_attr('alt-with-value'):
                tag['alt-with-value'] = '&quot;&quot;'
                
            if not tag.has_attr('filepath-with-value'):
                tag.decompose()
            else:
                if tag['filepath-with-value'] in empty_values:
                    tag.decompose()
                    
            if not tag.has_attr('caption-with-value'):
                tag['caption-with-value'] = '&quot;&quot;'
```

Please add empty lines between unrelated `if`, the code is then easier to read.

Reply:
ditto elsewhere

Reply:
Done